pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -317,29 +316,15 @@ func parseList(legacyList []runtime.Object) (metainternalversion.ListOptions, ma
 	indexMap := map[string]int{}
 
 	for i, obj := range legacyList {
-		metaAccessor, err := utils.MetaAccessor(obj)
-		if err != nil {
-			return options, nil, err
-		}
-		originKeys = append(originKeys, metaAccessor.GetOriginKey())
-
-		accessor, err := meta.Accessor(obj)
+		accessor, err := utils.MetaAccessor(obj)
 		if err != nil {
 			return options, nil, err
 		}
 		indexMap[accessor.GetName()] = i
 	}
-
 	if len(originKeys) == 0 {
 		return options, nil, nil
 	}
-
-	r, err := labels.NewRequirement(utils.AnnoKeyOriginKey, selection.In, originKeys)
```

Comment:
this logic is confusing too me -- we can not depend on this existing 😬  it may work for playlists, but not in general.

I think just checking name is sufficient

Reply:
This is attempting to filter the list of objects we're loading from US to just the objects in the list from legacy (to avoid paging issues mostly).  Assuming the names match between legacy and US you're absolutely right that we can just make a list of keys and pass those through to the Us list call.